widget/view.controller.js

```diff
@@ -366,49 +409,128 @@
         function nextNotification(threatHuntConfigForm) {
             if (!CommonUtils.isUndefined(threatHuntConfigForm.fazForm) && threatHuntConfigForm.fazForm.$invalid) {
                 var huntToolIndex = $scope.selectedEnv.huntTools.indexOf('Fortinet FortiAnalyzer');
-                _activeErrorTab(huntToolIndex, 'Fortinet FortiAnalyzer');
+                _activeErrorTab('Fortinet FortiAnalyzer', huntToolIndex);
                 toaster.error({
                     body: 'Fortinet FortiAnalyzer Threat Hunt Tool parameters are required'
                 });
                 return;
             } else if (!CommonUtils.isUndefined(threatHuntConfigForm.fsmForm) && threatHuntConfigForm.fsmForm.$invalid) {
                 var huntToolIndex = $scope.selectedEnv.huntTools.indexOf('Fortinet FortiSIEM');
-                _activeErrorTab(huntToolIndex, 'Fortinet FortiSIEM');
+                _activeErrorTab('Fortinet FortiSIEM', huntToolIndex);
                 toaster.error({
                     body: 'Fortinet FortiSIEM Threat Hunt Tool parameters are required'
                 });
                 return;
             }
             else if (!CommonUtils.isUndefined(threatHuntConfigForm.qradarForm) && threatHuntConfigForm.qradarForm.$invalid) {
                 var huntToolIndex = $scope.selectedEnv.huntTools.indexOf('IBM QRadar');
-                _activeErrorTab(huntToolIndex, 'IBM QRadar');
+                _activeErrorTab('IBM QRadar', huntToolIndex);
                 toaster.error({
                     body: 'IBM QRadar Threat Hunt Tool parameters are required'
                 });
                 return;
             } else if (!CommonUtils.isUndefined(threatHuntConfigForm.splunkForm) && threatHuntConfigForm.splunkForm.$invalid) {
                 var huntToolIndex = $scope.selectedEnv.huntTools.indexOf('Splunk');
-                _activeErrorTab(huntToolIndex, 'Splunk');
+                _activeErrorTab('Splunk', huntToolIndex);
                 toaster.error({
                     body: 'Splunk Threat Hunt Tool parameters are required'
                 });
                 return;
             } else {
-                $scope.selectedEnv.installOutbreakType = $scope.outbreakAlertSeverityList.slice();
-                WizardHandler.wizard('OutbreaksolutionpackWizard').next();
+                $scope.isConnectorsHealthy = true;
+                // Array to hold all promises
+                let promises = [];
```

Comment:
Can this be a part of any function? please break it down to a function or multiple functions

Reply:
done